Cubical/Algebra/Group/Abelianization/AbelianizationAsCoeq.agda

```diff
@@ -0,0 +1,354 @@
+       f   ∃! inducedHom
+         \   .
+           \ .
+             H
+  commuting diagram
+  -}
+  inducedHom : (H : AbGroup ℓ)
+             → (f : GroupHom G (AbGroup→Group H))
+             → AbGroupHom asAbelianGroup H
+  inducedHom H f = g , gIsHom
+    where open IsGroupHom
+          instance
+            _ = snd (AbGroup→Group H)
+          f' = fst f
+          g = rec
+                (isSetAbGroup H)
+                (λ x → (f') x)
+                (λ (a , b , c) → f' (a · b · c)     ≡⟨ (snd f).pres· a (b · c) ⟩
+                           (f' a) · (f' (b · c))    ≡⟨ cong (λ x → (f' a) · x) ((snd f).pres· b c) ⟩
+                           (f' a) · (f' b) · (f' c) ≡⟨ cong (λ x → (f' a) · x) ((snd H).AbGroupStr.comm (f' b) (f' c)) ⟩
+                           (f' a) · (f' c) · (f' b) ≡⟨ cong (λ x → (f' a) · x) (sym ((snd f).pres· c b)) ⟩
+                           (f' a) · (f' (c · b))    ≡⟨ sym ((snd f).pres· a (c · b)) ⟩
+                           f' (a · c · b) ∎)
+          gIsHom : IsGroupHom (snd (AbGroup→Group asAbelianGroup)) g (snd (AbGroup→Group H))
+          pres· gIsHom =
+            elimProp2
+              (λ x y → isSetAbGroup H _ _)
+              ((snd f).pres·)
+          pres1 gIsHom = (snd f).pres1
+          presinv gIsHom =
+            elimProp
+              (λ x → isSetAbGroup H _ _)
+              ((snd f).presinv)
+
+  commutativity : (H : AbGroup ℓ)
+                → (f : GroupHom G (AbGroup→Group H))
+                → (compGroupHom incAbAsGroupHom (inducedHom H f) ≡ f)
+  commutativity H f =
+      Σ≡Prop
+        (λ _ → isPropIsGroupHom _ _)
+        (λ i x → q x i)
+    where q : (x : fst  G)
+              → fst (compGroupHom incAbAsGroupHom (inducedHom H f)) x ≡ fst f x
+          q = (λ x → refl)
+
+  uniqueness : (H : AbGroup ℓ)
+             → (f : GroupHom G (AbGroup→Group H))
+             → (g : AbGroupHom asAbelianGroup H)
+             → (p : compGroupHom incAbAsGroupHom g ≡ f)
+             → (g ≡ inducedHom H f)
+  uniqueness H f g p =
+      Σ≡Prop
+        (λ _ → isPropIsGroupHom _ _)
+        (λ i x →  q x i)
+    where q : (x : Abelianization)
+              →  fst g x ≡ fst (inducedHom H f) x
+          q = elimProp
+                (λ _ → isSetAbGroup H _ _)
+                (λ x → fst g (incAb x) ≡⟨ cong (λ f → f x) (cong fst p) ⟩
+                       (fst f) x       ≡⟨ refl ⟩
+                       fst (inducedHom H f) (incAb x)∎)
+
+  {- The proof that defining the abelianization of groups as a coequalizer of sets is equivalent
+  to defining it as a HIT, more precisely that there is an isomorphism between the resulting
+  abelian groups unique with respect to commuting with the constructors η and inc.
+
+  G -incAb-> abelianization as coequalizer
+   \         .
+     \       .
+       η   ∃! isomorphism
+         \   .
+           \ .
+             abelianization as HIT
+  commuting diagram
+  -}
+  isomorphism : AbGroupIso asAbelianGroup (HITasAbelianGroup G)
```

Comment:
This could be exported directly, if you name it more specifically.

Reply:
is this still necessary, with the proof being its own module? i feel like i should then export the other two lemmas too. i am open for renaming it though

Reply:
exported the isomorphism directly